arch/RISCV/RISCVInstPrinter.c

```diff
@@ -0,0 +1,183 @@
+/// from the register set description.  This returns the assembler name
+/// for the specified register.
+static const char *getRegisterName(unsigned RegNo, unsigned AltIdx);
+
+// Include the auto-generated portion of the assembly writer.
+#define PRINT_ALIAS_INSTR
+#include "RISCVGenAsmWriter.inc"
+
+/// NULL
+void RISCV_post_printer(csh ud, cs_insn *insn, char *insn_asm, MCInst *mci) 
+{
+  	assert(0 && "RISCV post-printer doesn't implementated.");
+	/*
+	   if (((cs_struct *)ud)->detail != CS_OPT_ON)
+	   return;
+	 */
+}
+
+//void RISCVInstPrinter::printInst(const MCInst *MI, raw_ostream &O,
+//                                 StringRef Annot, const MCSubtargetInfo &STI) 
+void RISCV_printInst(MCInst *MI, SStream *O, void *info) 
+{
+  	MCRegisterInfo *MRI = (MCRegisterInfo *) info;
+  	//bool Res = false;
+  	MCInst *NewMI = MI;
+  	// TODO: RISCV compressd instructions.
+  	//MCInst UncompressedMI;
+  	//if (!NoAliases)
+    	//Res = uncompressInst(UncompressedMI, *MI, MRI, STI);
+  	//if (Res)
+    	//NewMI = const_cast<MCInst *>(&UncompressedMI);
+  	if (/*NoAliases ||*/ !printAliasInstr(NewMI, O, info))
+    		printInstruction(NewMI, O, MRI);
+  		//printAnnotation(O, Annot);
+}
+
+static void printRegName(SStream *OS, unsigned RegNo) 
+{
+  	SStream_concat0(OS, getRegisterName(RegNo, RISCV_ABIRegAltName));
+}
+
+/**
+void RISCVInstPrinter::printOperand(const MCInst *MI, unsigned OpNo,
+                                    raw_ostream &O, const char *Modifier) 
+*/
+static void printOperand(MCInst *MI, unsigned OpNo, SStream *O) 
+{
+  	unsigned reg;
+  	int64_t Imm = 0;
+
+  	MCOperand *MO = MCInst_getOperand(MI, OpNo);
+  
+  	//
+  	if (MCOperand_isReg(MO)) {
+    		reg = MCOperand_getReg(MO);
+    		printRegName(O, reg);
+    		if (MI->csh->detail) {
+      			MI->flat_insn->detail->riscv.operands[MI->flat_insn->detail->riscv.op_count].type = RISCV_OP_REG;
+      			MI->flat_insn->detail->riscv.operands[MI->flat_insn->detail->riscv.op_count].reg = reg;
+      			MI->flat_insn->detail->riscv.op_count++;
+    		}
+
+    		return;
+  	}
+
+  	//
+  	if (MCOperand_isImm(MO)) {
+    		Imm = MCOperand_getImm(MO);
+    		if (Imm >= 0) {
+      			if (Imm > HEX_THRESHOLD)
+        			SStream_concat(O, "0x%" PRIx64, Imm);
+      			else
+				SStream_concat(O, "%" PRIu64, Imm);
+    		} else {
+      			if (Imm < -HEX_THRESHOLD)
+				SStream_concat(O, "-0x%" PRIx64, -Imm);
+      			else
+				SStream_concat(O, "-%" PRIu64, -Imm);
+    		}
+
+    		if (MI->csh->detail) {
+      			MI->flat_insn->detail->riscv.operands[MI->flat_insn->detail->riscv.op_count].type = RISCV_OP_IMM;
+      			MI->flat_insn->detail->riscv.operands[MI->flat_insn->detail->riscv.op_count].imm = Imm;
+      			MI->flat_insn->detail->riscv.op_count++;
+    		}
+    		return;
+  	}
+
+  	//assert(MO.isExpr() && "Unknown operand kind in printOperand");
+  	assert(0 && "Unknown operand kind in printOperand");
```

Comment:
this assert line can be removed? then the check for Imm operand type on line 112 can be removed, too.

Reply:
This is old code from PR1198, looks like can be removed.